static/app/components/forms/controls/rangeSlider/index.tsx

```diff
@@ -219,6 +221,12 @@ function RangeSlider({
   );
 }
 
+const StyledSlider = styled(Slider, {
+  shouldForwardProp: prop => isPropValid(prop),
+})<{hasLabel: boolean}>`
+  margin: ${space(1)} 0 ${p => space(p.hasLabel ? 2 : 1)};
+`;
```

Comment:
It’s quite confusing to me that all the `hasLabel` prop does is add some extra margin. There seems to also be only one usage of that prop, where we tie it to `showCustomInput`.

Maybe we could instead put a margin-top on the custom input instead of a margin-bottom on the Slider itself?

Reply:
I think so yes, but it seems that we still have some default left margin here, meaning we'd probably end up with two different styled elements. I'll change it to that.

Reply:
this is another argument for me that layout component are actually good. You have no way of knowing how those components will look in the UI if you just `Slider` and `Input` being rendered as parent-child, and one of those has a margin somewhere 🤷